eli5/shap_xgboost.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from functools import partial
+import re
+from typing import Any, Dict, List, Tuple, Optional, Pattern
+
+import numpy as np  # type: ignore
+import scipy.sparse as sp  # type: ignore
+from xgboost import (  # type: ignore
+    XGBClassifier,
+    XGBRegressor,
+    Booster,
+    DMatrix
+)
+
+from eli5.explain import explain_weights, explain_prediction
+from eli5.sklearn.utils import (
+    add_intercept,
+    get_X,
+    get_X0,
+    handle_vec,
+    predict_proba
+)
+from eli5.utils import is_sparse_vector
+from eli5._decision_path import get_decision_path_explanation
+from eli5._feature_importances import get_feature_importance_explanation
+from eli5._shap_explanation_summary import get_shap_explanation_summary
+
+DESCRIPTION_XGBOOST_SHAP = """
+XGBoost feature importances; values are numbers 0 <= x <= 1;
+all values sum to 1.
+"""
+
+def explain_shap_prediction_xgboost(
+        xgb, doc,
+        vec=None,
+        top=None,
+        top_targets=None,
+        target_names=None,
+        targets=None,
+        feature_names=None,
+        feature_re=None,  # type: Pattern[str]
+        feature_filter=None,
+        vectorized=False,  # type: bool
+        is_regression=None,  # type: bool
+        missing=None,  # type: bool
+        ):
+    
+    booster, is_regression = _check_booster_args(xgb)
+    xgb_feature_names = booster.feature_names
+    vec, feature_names = handle_vec(
+        xgb, doc, vec, vectorized, feature_names,
+        num_features=len(xgb_feature_names))
+    
+    #adding the intercept to the feature names. 
+    feature_names.add_feature('<BIAS>')
+    
+    if missing is None:
+        missing = np.nan if isinstance(xgb, Booster) else xgb.missing
+    
+    if str(type(doc)).endswith("xgboost.core.DMatrix'>"):
+        dmatrix = doc
+    else:
+        if(vec is not None):
+            dmatrix=DMatrix(get_X(doc,vec=vec,vectorized=vectorized), missing=missing)
+        else:
+            dmatrix = DMatrix(doc, missing=missing)
+    coef = _xgb_feature_importances(booster, dmatrix)
+    return get_shap_explanation_summary(xgb, vec=vec, coef=coef,
```

Comment:
I think I said you the other day something wrong. I said that you shouldn't use `get_decision_path_explanation`, but I was wrong. Probably you should use it here instead of `get_shap_explanation_summary` as was used here: https://github.com/AshwinB-hat/eli5/blob/master/eli5/xgboost.py#L201. Some ideas (I could be wrong):
- `explain_shap_prediction` in eli5 only supports a single sample in doc. It is a limitation of Eli5. It would be nice to support explanations of multiple documents at the same time, but forget about that by now.
- `score_weights` (https://github.com/AshwinB-hat/eli5/blob/master/eli5/xgboost.py#L195) in your case would be a mix between the output of `_xgb_feature_importances` and the predictions for the document, but the prediction for the document is the sum of the feature importances according to https://github.com/dmlc/xgboost/blob/29a1356669cb3c3ca2a6b22bced5047912072804/tests/python/test_shap.py#L64, so much easier, just sum all the contributions for each class for the scores in `score_weights`, but before confirm that I'm right and that the sum of the contributions is equal to the prediction for the document.
- Probably the rest could keep being the same (not sure). If that is the case, then what you would have to do is just replace this two lines https://github.com/AshwinB-hat/eli5/blob/master/eli5/xgboost.py#L195-L196 with some code based in `_xgd_feature_importances` that gives the desired `score_weights`. After that, all existing test should keep working. 

Sorry for saying you something wrong the other day. Hope now I have said something that makes more sense :-)

Reply:
@ivanprado ,I have integrated xgboost. Over the next two days, I will try to finish `catboost` and `lightgbm`